### DISCUSSION TITLE
Sdf.Layer.CreateNew() hangs maya 2020 on exit

### DISCUSSION THREAD

**Discussion by DanEnglesson** [Autodesk/maya-usd#1270] [Q&A]
Hi,

I'm testing out USD 20.11 in Maya 2020 on Windows 10 and stuff seems to work nicely apart from when closing the Maya session. When trying to close Maya it just hangs and you need to force close it.

I've been able to narrow it down to Maya not releasing one of my layers properly. If I do a Sdf.Layer.CreatNew() call Maya seems to take ownership of the layer pointer. I would expect to be able to release the layer when I'm done with it and go on my merry way, but seems like Maya still keeps it. I would expect to be able to do the following simple test in the python script editor.

**Example:**
Run this in Maya 2020's script editor and then quit Maya, it hangs and I have to kill the process manually.
```
from pxr import Sdf
root_layer = Sdf.Layer.CreateNew("test_out.usd")
root_layer.Save()
root_layer = None
# printing the loaded layers after this I don't have any in the Maya session
print Sdf.Layer.GetLoadedLayers()
```

The output I'm getting is showing that the layer is removed but still Maya hangs on exit. Is this expected?
TF_DEBUG Output:
```
...
Sdf_ComputeAssetInfoFromIdentifier:
  assetInfo->identifier = 'c:/Users/dan.englesson/test_out.usd'
  assetInfo->realPath = 'C:\Users\dan.englesson\test_out.usd'
  assetInfo->repoPath = ''
  assetInfo->assetName = ''
  assetInfo->version = ''
Sdf_LayerRegistry::InsertOrUpdate(SdfLayer('c:/Users/dan.englesson/test_out.usd', 'C:\Users\dan.englesson\test_out.usd'))
SdfLayer::~SdfLayer('c:/Users/dan.englesson/test_out.usd')
Sdf_LayerRegistry::Erase(SdfLayer('c:/Users/dan.englesson/test_out.usd', 'C:\Users\dan.englesson\test_out.usd')) => Success
```

**Environment:**
**USD:** 20.11.0
**MayaUSD:** 0.5.0
**Maya:** 2020
**OS:** Windows 10

Has anyone run into the same issue or something similar?
Any help would be much appreciated.
Many thanks,
Dan

**Comment by kxl-adsk** [comment]
Hi Dan, I tried to reproduce your issue with what's currently in the dev branch + Maya 2020 + USD 20.11 on Windows and I'm not getting any hangs. Would it be possible for you to try the latest version of the plugin?

Cheers,
KxL

**Reply by DanEnglesson** [reply]
Thanks for testing and verifying it works for you. I'll test against the dev-branch.

Cheers,
Dan
